RXPatchLib/WebPatchSource.cs

```diff
@@ -152,35 +216,31 @@ private void UnlockDownload()
                     }
                     catch (HashMistmatchException)
                     {
+                        RxLogger.Logger.Instance.Write($"Invalid file hash for {subPath} - Expected hash {hash}, requeuing download");
+
                         // Try the next best host; throw an exception if there is none
-                        if (Patcher.PopHost() == null)
+                        if (_patcher.PopHost() == null)
                             throw new NoReliableHostException();
 
                         // Reset progress and requeue download
                         await LoadNew(subPath, hash, cancellationToken, progressCallback);
                     }
-                }
-            }
+                    catch (WebException)

```

Comment:
We shouldn't actually be getting `WebException` here; the retry strategy should be catching those and eventually throwing TooManyRetriesException.